rust/src/jit_proxy_client.rs

```diff
@@ -0,0 +1,185 @@
+            bid,
+            ask,
+            price_type,
+            referrer_info,
+            post_only,
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct JitProxyClient<T: AccountProvider> {
+    drift_client: DriftClient<T>,
+    config: Option<RpcSendTransactionConfig>,
+    cu_params: Option<ComputeBudgetParams>,
+}
+
+
+impl<T: AccountProvider> JitProxyClient<T> {
+    pub fn new(drift_client: DriftClient<T>, config: Option<RpcSendTransactionConfig>, cu_params: Option<ComputeBudgetParams>) -> Self {
+        Self {
+            drift_client,
+            config,
+            cu_params
+        }
+    }
+
+    pub fn update_config(&mut self, config: RpcSendTransactionConfig) {
+        self.config = Some(config);
+    }
+
+    pub fn update_cu_params(&mut self, cu_params: ComputeBudgetParams) {
+        self.cu_params = Some(cu_params);
+    }
+
+    pub async fn jit(&self, params: JitIxParams) -> JitResult<Signature> {        
+        if let Some(order) = params.taker.orders.iter().find(|order| order.order_id == params.taker_order_id) { 
+            let tx_builder = self.drift_client.init_tx(&self.drift_client.wallet().default_sub_account(), false).await.unwrap();
+            let program_data = tx_builder.program_data();
+            let account_data = tx_builder.account_data();
+            
+            let wallet =  self.drift_client.wallet();
+
+            let writable_markets = match order.market_type {
+                MarketType::Perp => {
+                    vec![MarketId::perp(order.market_index)]
+                }
+                MarketType::Spot => {
+                    vec![MarketId::spot(order.market_index), MarketId::QUOTE_SPOT]
+                }
+            };
+
+            let mut accounts = build_accounts(
+                program_data,
+                jit_proxy::accounts::Jit {
+                    state: *state_account(),
+                    user: wallet.default_sub_account(),
+                    user_stats: drift_sdk::Wallet::derive_stats_account(wallet.authority(), &PROGRAM_ID),
+                    taker: params.taker_key,
+                    taker_stats: params.taker_stats_key,
+                    authority: *wallet.authority(),
+                    drift_program: PROGRAM_ID,
+                },
+                &[&params.taker, account_data],
+                &[],
+                &writable_markets.as_slice(),
+            );
+
+            if let Some(referrer_info) = params.referrer_info {
+                accounts.push(AccountMeta::new(referrer_info.referrer(), false));
+                accounts.push(AccountMeta::new(referrer_info.referrer_stats(), false));
+            }
+
+            if order.market_type == MarketType::Spot {
+                let spot_market_vault = self.drift_client.get_spot_market_info(order.market_index).await?.vault;
+                let quote_spot_market_vault = self.drift_client.get_spot_market_info(QUOTE_SPOT_MARKET_INDEX).await?.vault;
```

Comment:
can use tokio::join! to make this concurrent

```suggestion
                let (spot_market,  quote_spot_market ) = tokio::join!(self.drift_client.get_spot_market_info(order.market_index), self.drift_client.get_spot_market_info(QUOTE_SPOT_MARKET_INDEX));
```